### PR TITLE
Expose functions to get Advertising ID

### PR DESCRIPTION
**Pull request by dhawal1248**
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-android/38)

### CODE REVIEW

File: core/src/main/java/com/rudderstack/android/sdk/core/RudderContext.java [hunk 1]
```diff
@@ -185,4 +185,9 @@ private boolean getAmazonFireAdvertisingID() throws Exception {
         return true;
     }
 
+    public String getAdvertisingId() {
```
Comment (arnab-p) [inline core/src/main/java/com/rudderstack/android/sdk/core/RudderContext.java:188]: Add Javadoc and @Nullable

File: core/src/main/java/com/rudderstack/android/sdk/core/RudderContext.java [hunk 2]
```diff
@@ -185,4 +185,9 @@ private boolean getAmazonFireAdvertisingID() throws Exception {
...
+        if (this.deviceInfo == null) { return null; }
+        return this.deviceInfo.getAdvertisingId();
+    }
+
```
Comment (arnab-p) [inline core/src/main/java/com/rudderstack/android/sdk/core/RudderContext.java:192]: Expose the `adTrackingEnabled` as well?

Comment (dhawal1248) [reply core/src/main/java/com/rudderstack/android/sdk/core/RudderContext.java:198]: done

File: core/src/main/java/com/rudderstack/android/sdk/core/RudderContext.java (hunk 1 above)
Comment (dhawal1248) [reply core/src/main/java/com/rudderstack/android/sdk/core/RudderContext.java]: done

File: core/src/main/java/com/rudderstack/android/sdk/core/RudderContext.java (hunk 2 above)
Comment (arnab-p) [reply core/src/main/java/com/rudderstack/android/sdk/core/RudderContext.java:198]:
```suggestion
@Nullable public String get...
```

Check : https://github.com/rudderlabs/rudder-sdk-android/blob/07484c210c252b3c07320654a7185155e3cf8238/core/src/main/java/com/rudderstack/android/sdk/core/RudderClient.java#L40

**Pull request merged**